store/views.py

```diff
@@ -1,38 +1,44 @@
-from django.shortcuts import render
+from django.shortcuts import render,redirect
 from django.shortcuts import get_object_or_404
+from django.shortcuts import get_list_or_404
+from datetime import date
 from store.models import *
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
+from decimal import Decimal
 
-# Create your views here.
 
 def index(request):
     return render(request, 'store/index.html')
 
 def bookDetailView(request, bid):
     template_name = 'store/book_detail.html'
     context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
-    }
-    # START YOUR CODE HERE
-    
+        'book': None,
+        'num_available': None,
+    }   
     
+    book = get_object_or_404(Book,pk=bid)
+    bookcopy = get_list_or_404(BookCopy, book=bid, status=True)
```

Comment:
Even if the bookcopy does not exist, the book detail can be viewed by the user. Only he cannot loan the book.